src/hooks/useQuota/useQuota.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState, useContext, useCallback, useEffect } from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { getQuota, NotEligibleError } from "../../services/quota";
+import { ProductContext } from "../../context/products";
+import { transform } from "lodash";
+import { Quota, CampaignPolicy } from "../../types";
+import { usePrevious } from "../usePrevious";
+
+type QuotaState = "DEFAULT" | "FETCHING_QUOTA" | "NO_QUOTA" | "NOT_ELIGIBLE";
+
+export type QuotaHook = {
+  quotaState: string;
+  // Contains only quota of products with categories available in the current campaign policy
+  quotaResponse?: Quota;
+  // Contains quota of all products
+  allQuotaResponse?: Quota;
+  updateQuota: () => void;
+  quotaError?: Error;
+};
+
+const filterQuotaWithAvailableProducts = (
+  quota: Quota,
+  products: CampaignPolicy[]
+): Quota => {
+  const filteredQuota: Quota = {
+    remainingQuota: [],
+    globalQuota: [],
+    localQuota: []
+  };
+
+  transform(
+    quota.remainingQuota,
+    (result: Quota, itemQuota) => {
+      if (products.some(policy => policy.category === itemQuota.category))
+        result.remainingQuota.push(itemQuota);
+    },
+    filteredQuota
+  );
+
+  transform(
+    quota.globalQuota!,
```

Comment:
Noticed the assertion, guess we don't need this anymore? EDIT: noticed there are a couple of assertions elsewhere too, perhaps can be removed too